Sonar fixes robert

robertkrantz opened the pull request.

Commit:
Removed use of deprecated method

Commit:
Removed use of deprecated method

Commit:
Removed unused import

Commit:
Added test for prePersist method and fixed null check

Commit:
Updated uvms-pom version to latest release.

Also had to updated some tests since mockito started complaining about unused stubs.

Commit:
Replaced deprecated methods.

Commit:
Replaced deprecated methods and removed unused members.

RobinMilding reviewed (approved):
Good cleanup

Commit:
Replaced deprecated methods and updated test so that it actually does something.

Commit:
Replaced deprecated class

Commit:
Replaced deprecated methods

Commit:
Replaced deprecated methods

Commit:
Replaced deprecated methods

RobinMilding commented on application/src/test/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/GearCharacteristicsMapperTest.java:
```diff
@@ -47,10 +49,9 @@ public void testMapGearDtoToFishingGearEntityWithTypeCodeDG(GearCharacteristicEn
-                $(entity, GearCharacteristicConstants.GEAR_CHARAC_TYPE_CODE_ME, GearDto.builder().meshSize("20.25kg").build())
-                // TODO TEST ALL CASES
-        );
+        return Arrays.asList(new Object[][] {
```
I tried it, you don't have to wrap the Object[][] in a List

Commit:
Fixed broken tests and removed unnecessary mocks

Commit:
Replaced depcrecated methods

Commit:
Simplified parameter methods

Commit:
Removed the catch endpoint.

Main reason for removing this is that it is not used and the implementation is very complicated for what it claims to do. The easier path is to remove it and when we need to extract catch data we can implement a much easier endpoint.

RobinMilding commented on application/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/facatch/FACatchSummaryHelper.java:
```diff
@@ -58,55 +58,57 @@ This file is part of the Integrated Fisheries Data Management (IFDM) Suite. The
 
-        Object faCatchSummaryCustomEntityObj = cls.newInstance();
-        Class parameterType = String.class;
+            Map<GroupCriteria, GroupCriteriaMapper> groupMappings = FilterMap.getGroupByMapping();
```
I'm happy that I won't stumble over this horrible indentation again from now on

RobinMilding reviewed (approved):
Looking forward to see what happens in Sonar after this merge!
